src/Compilers/Core/Portable/CodeGen/LocalSlotManager.cs

```diff
@@ -58,8 +58,8 @@ public bool Equals(LocalSignature other)
             public override int GetHashCode()
                 => Hash.Combine(_type, (int)_constraints);
 
-            public override bool Equals(object obj)
-                => Equals((LocalSignature)obj);
+            public override bool Equals(object? obj)
+                => obj is LocalSignature ls && Equals(ls);
```

Comment:
Would cast to `(LocalSignature?)` also have worked? Seems simpler.

Reply:
That only works when the argument to `Equals` is `null` or `LocalSignature`. The contract of `Equals` though is it must behave correctly even when incorrect types are specified.